clients/gtest/syr_gtest.cpp

```diff
@@ -7,14 +7,80 @@
 #include "rocblas_datatype2string.hpp"
 #include "rocblas_test.hpp"
 #include "testing_syr.hpp"
+#include "testing_syr_batched.hpp"
+#include "testing_syr_strided_batched.hpp"
 #include "type_dispatch.hpp"
 #include <cctype>
 #include <cstring>
 #include <type_traits>
 
 namespace
 {
-    // By default, this test does not apply to any types.
+    // possible test cases
+    enum syr_test_type
+    {
+        SYR,
+        SYR_BATCHED,
+        SYR_STRIDED_BATCHED,
+    };
+
+    //syr test template
+    template <template <typename...> class FILTER, syr_test_type SYR_TYPE>
+    struct syr_template : RocBLAS_Test<syr_template<FILTER, SYR_TYPE>, FILTER>
+    {
+        // Filter for which types apply to this suite
+        static bool type_filter(const Arguments& arg)
+        {
+            return rocblas_simple_dispatch<syr_template::template type_filter_functor>(arg);
+        }
+
+        // Filter for which functions apply to this suite
+        static bool function_filter(const Arguments& arg)
+        {
+            switch(SYR_TYPE)
+            {
+            case SYR:
+                return !strcmp(arg.function, "syr") || !strcmp(arg.function, "syr_bad_arg");
+            case SYR_BATCHED:
+                return !strcmp(arg.function, "syr_batched")
+                       || !strcmp(arg.function, "syr_batched_bad_arg");
+            case SYR_STRIDED_BATCHED:
+                return !strcmp(arg.function, "syr_strided_batched")
+                       || !strcmp(arg.function, "syr_strided_batched_bad_arg");
+            }
+            return false;
+        }
+
+        // Google Test name suffix based on parameters
+        static std::string name_suffix(const Arguments& arg)
+        {
+            RocBLAS_TestName<syr_template> name;
+
+            name << rocblas_datatype2string(arg.a_type);
+
+            if(strstr(arg.function, "_bad_arg") != nullptr)
+            {
+                name << "_bad_arg";
+            }
+            else
+            {
+                name << '_' << (char)std::toupper(arg.uplo) << '_' << arg.N << '_' << arg.alpha
+                     << '_' << arg.incx << '_' << arg.lda;
```

Comment:
I think we would want to keep the order of parameters in the name consistent with the argument list in the API for better readability when debugging. i.e.  incx << stride_x << lda << strideA << batch_count

Reply:
Sure thing I'll make it consistent to others, but then comparing non batched to batched then isn't easy via filter.